datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -1426,6 +1542,141 @@ pub fn unnest(input: LogicalPlan, column: Column) -> Result<LogicalPlan> {
     }))
 }
 
+// Create function name for unnest
+// Different from create_function_physical_name, so we create a customize one.
+// We need fun(arg1,arg2,arg3) format, but not fun(arg1, arg2, arg3).
+// TODO: We dont need this if we can customize format in impl::Display for Expr
+fn create_unnest_arguments_name(array_exprs: Vec<Expr>) -> Result<Vec<String>> {
+    array_exprs
+        .iter()
+        .map(|e| match e {
+            Expr::ScalarFunction(ScalarFunction { fun, args, .. }) => {
+                let arg_str = args
+                    .iter()
+                    .map(|e| e.to_string())
+                    .collect::<Vec<String>>()
+                    .join(",");
+
+                Ok(format!("{fun}({arg_str})"))
+            }
+            Expr::Literal(sv) => {
+                let name = format!("{sv:?}");
+                Ok(name)
+            }
+            e => Err(DataFusionError::NotImplemented(format!(
+                "Expr {e} is not supported in UNNEST"
+            ))),
+        })
+        .collect::<Result<Vec<String>>>()
+}
+
+/// Create unnest plan from arrays.
+fn build_unnest_plans(
```

Comment:
I am surprised this can't use https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.LogicalPlan.html#variant.Unnest